src/cdk/v2/destinations/intercom/procWorkflow.yaml

```diff
@@ -0,0 +1,170 @@
+      payload.user_id = !(payload.user_id) && .destination.Config.sendAnonymousId && .message.anonymousId ? .message.anonymousId : payload.user_id;
+      payload;
+
+  - name: prepareTrackPayload
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.context.payload = .message.context.mappedToDestination === true ? $.outputs.rEtlPayload : $.outputs.trackEtlPayload;
+      $.context.requestMethod = 'POST';
+      $.context.endpoint =  $.getBaseEndpoint(.destination) + "/" + "events";
+      $.context.payload = $.removeUndefinedAndNullValues($.context.payload);
+
+  - name: validateTrackPayload
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.assert($.context.payload.event_name, "Event name is required for track call");
+      $.assert($.context.payload.user_id || $.context.payload.email, "Either email or userId is required for Track call");
+
+  - name: groupEtlPayload
+    condition: $.outputs.messageType === {{$.EventType.GROUP}} && !.message.context.mappedToDestination
+    template: |
+      const payload = .message.({
+        company_id:  {{{{$.getGenericPaths("groupId")}}}},
+        name: {{{{$.getGenericPaths("name")}}}},
+        website: {{{{$.getGenericPaths("website")}}}},
+        plan: .traits.plan || .context.traits.plan,
+        size: Number(.traits.size || .context.traits.size),
+        industry: .traits.industry || .context.traits.industry,
+        monthly_spend: .traits.monthlySpend || .context.traits.monthlySpend ? Number(.traits.monthlySpend || .context.traits.monthlySpend) : undefined,
+        remote_created_at: .traits.remoteCreatedAt || .context.traits.remoteCreatedAt ? Number(.traits.remoteCreatedAt || .context.traits.remoteCreatedAt) : undefined
+      })
+      payload.custom_attributes = .message.traits || {};
+      payload.custom_attributes = $.filterCustomAttributes(payload, "company");
+      payload;
+
+  - name: prepareAddUserToCompanyPayload
+    condition: $.outputs.messageType === {{$.EventType.GROUP}} && $.isDefinedAndNotNull($.outputs.searchContact)
+    template: |
+      $.assert(.message.groupId, "groupId is required for group call");
+      let payload = .message.context.mappedToDestination === true ? $.outputs.rEtlPayload : $.outputs.groupEtlPayload;
+      const contactId = $.outputs.searchContact;
+      const companyId = await $.createOrUpdateCompany(payload, .destination);
+      $.assert(companyId, "Unable to add user to company");
+      payload.requestBodyJson = {
+        id: companyId
+      }
+      payload.endpoint =  $.getBaseEndpoint(.destination) + "/" + "contacts" + "/" + contactId + "/" + "companies";
+      payload;
+
+  - name: prepareCreateOrUpdateCompanyPayload
+    condition: $.outputs.messageType === {{$.EventType.GROUP}} && !$.isDefinedAndNotNull($.outputs.searchContact)
+    template: |
+      $.assert(.message.groupId, "groupId is required for group call");
```

Comment:
This is redundant.

Reply:
this is required because groupId is mapped with company_id and intercom does lookup based on company_id. so it's always recommended to send groupId